internal/services/redis/redis_cache_data_source.go

```diff
@@ -344,6 +349,12 @@ func dataSourceRedisCacheRead(d *pluginsdk.ResourceData, meta interface{}) error
 		d.Set("primary_connection_string", getRedisConnectionString(*props.HostName, *props.SslPort, *keys.Model.PrimaryKey, enableSslPort))
 		d.Set("secondary_connection_string", getRedisConnectionString(*props.HostName, *props.SslPort, *keys.Model.SecondaryKey, enableSslPort))
 
+		accessKeyAuthEnabled := true
+		if props.DisableAccessKeyAuthentication != nil {
+			accessKeyAuthEnabled = !(*props.DisableAccessKeyAuthentication)
+		}
+		d.Set("access_keys_authentication_enabled", accessKeyAuthEnabled)
```

Comment:
It's a common pattern in the provider to make use of `pointer.From` when setting properties that are pointers into state. `pointer.From` will nil check the input and if the pointer is nil will return the zero value for the type that was passed in, so this would be condensed down to:
```suggestion
		d.Set("access_keys_authentication_enabled", !pointer.From(props.DisableAccessKeyAuthentication))
```

Reply:
@stephybun is right, I definitely missed this out.

Reply:
Ah nice idea, thx @stephybun . I will alter the suggested code a bit: we want to make the expression defaults to `true` if we get nil value on `DisableAccessKeyAuthentication`.

Reply:
Addressed in 50de57ee58

Reply:
@gerrytan that is the outcome of my suggestion above. 

If `props.DisableAccessKeyAuthentication` is `nil`, `pointer.From` will return `false` (the zero value for bools) which will result in `true` since it's being inverted. The variable initialization and nil check on lines 357-361 are redundant. Can you please apply the suggestion as made?

Reply:
@stephybun ah I get it now. Fixed in 5341f2f58f.